File: examples/src/main/java/org/rajawali3d/examples/ExamplesFragment.java
package org.rajawali3d.examples;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.appcompat.widget.SearchView;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import org.rajawali3d.examples.data.DataSet;
import org.rajawali3d.examples.recycler.CategoryAdapter;

public class ExamplesFragment extends Fragment implements SearchView.OnQueryTextListener {

    static final String TAG = "ExamplesFragment";

    private CategoryAdapter adapter;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);

        adapter = new CategoryAdapter();
        adapter.setCategories(DataSet.getInstance().getCategories());
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_examples, container, false);
        RecyclerView recyclerView = view.findViewById(R.id.recycler);
        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        recyclerView.setAdapter(adapter);

        return view;
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        inflater.inflate(R.menu.menu_search, menu);

        MenuItem menuItemSearch = menu.findItem(R.id.action_search);
        SearchView searchView = (SearchView) menuItemSearch.getActionView();
        searchView.setOnQueryTextListener(this);
    }

    @Override
    public boolean onQueryTextSubmit(String query) {
        return adapter.filterDone(requireActivity());
    }

    @Override
    public boolean onQueryTextChange(String newText) {
        adapter.filter(requireActivity(), newText);
        adapter.notifyDataSetChanged();
        return true;
    }

}

File: examples/src/main/java/org/rajawali3d/examples/ExamplesFragment.kt
package org.rajawali3d.examples

import android.content.Intent
import android.os.Bundle
import android.view.*
import androidx.appcompat.widget.SearchView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import org.rajawali3d.examples.data.Example
import org.rajawali3d.examples.data.ExamplesDataSet.Companion.instance
import org.rajawali3d.examples.examples.general.BasicFragment
import org.rajawali3d.examples.recycler.CategoryAdapter
import java.time.Clock.system

class ExamplesFragment : Fragment(), SearchView.OnQueryTextListener {

    private lateinit var adapter: CategoryAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
        adapter = CategoryAdapter()
        adapter.setCategories(instance!!.categories)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_examples, container, false)
        val recyclerView = view.findViewById<RecyclerView>(R.id.recycler)
        recyclerView.layoutManager = LinearLayoutManager(activity)
        recyclerView.adapter = adapter
        return view
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.menu_search, menu)
        val menuItemSearch = menu.findItem(R.id.action_search)
        val searchView = menuItemSearch.actionView as SearchView?
        searchView!!.setOnQueryTextListener(this)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_info -> {
                val intent = Intent(context, ExamplesActivity::class.java)
                intent.putExtra(ExamplesActivity.EXTRA_EXAMPLE, Example(R.string.system, SystemInfoFragment::class.java))
                startActivity(intent)
            }
        }
        return super.onOptionsItemSelected(item)
    }
    override fun onQueryTextSubmit(query: String) = adapter.filterDone(requireActivity())

    override fun onQueryTextChange(newText: String): Boolean {
        adapter.filter(requireActivity(), newText)
        adapter.notifyDataSetChanged()
        return true
    }

    companion object {
        const val TAG = "ExamplesFragment"
    }
}